pg_search/src/api/operator.rs

```diff
@@ -15,74 +15,237 @@
+        opretset: false,
+        opcollid: pg_sys::DEFAULT_COLLATION_OID,
+        inputcollid: pg_sys::DEFAULT_COLLATION_OID,
+        args: std::ptr::null_mut(),
+        location: (*(*srs).fcall).location,
     };
 
-    hash_set.contains(&key_field_value)
+    // we need to use what should be the only `USING bm25` index on the table
+    let rte = planner_rt_fetch((*var).varno as pg_sys::Index, (*srs).root);
+    let heaprel = PgRelation::open((*rte).relid);
+    let indexrel = locate_bm25_index((*rte).relid).unwrap_or_else(|| {
+        panic!(
+            "relation `{}.{}` must have a `USING bm25` index",
+            heaprel.namespace(),
+            heaprel.name()
+        )
+    });
+
+    let heaptupdesc = heaprel.tuple_desc();
+    let varatt = heaptupdesc
+        .get(
+            ((*var).varattno - 1)
+                .try_into()
+                .expect("varattno should be within the bounds of usize"),
+        )
+        .expect("varattno should exist in the heap");
+
+    // fabricate a `SearchConfig` from the above relation and query string
+    // and get it serialized into a JSONB Datum
+    let search_config = SearchConfig::from((query, indexrel));
+    if search_config.key_field != varatt.name() {
+        panic!("left-hand side of the @@@ operator should be the `key_field` of the only `USING bm25` index");
+    }
+
+    let search_config_json =
+        serde_json::to_value(&search_config).expect("SearchConfig should serialize to json");
+    let jsonb_datum = JsonB(search_config_json).into_datum().unwrap();
+
+    // from which we'll create a new pg_sys::Const node
+    let jsonb_const = pg_sys::makeConst(
+        pg_sys::JSONBOID,
+        -1,
+        pg_sys::DEFAULT_COLLATION_OID,
+        -1,
+        jsonb_datum,
+        false,
+        false,
+    );
+
+    // and assign it to the original argument list
+    input_args.replace_ptr(1, jsonb_const.cast());
+
+    // then assign that list to our new OpExpr node
+    newopexpr.args = input_args.as_ptr();
+
+    // copy that node into the current memory context and return it
+    let node = PgMemoryContexts::CurrentMemoryContext
+        .copy_ptr_into(&mut newopexpr, size_of::<pg_sys::OpExpr>());
+
+    ReturnedNodePointer(NonNull::new(node.cast()))
 }
 
 extension_sql!(
     r#"
+ALTER FUNCTION paradedb.search_with_search_config SUPPORT paradedb.search_config_support;
+ALTER FUNCTION paradedb.search_with_text SUPPORT paradedb.text_support;
+ALTER FUNCTION paradedb.search_with_query_input SUPPORT paradedb.query_input_support;
+
+CREATE OPERATOR pg_catalog.@@@ (
+    PROCEDURE = search_with_search_config,
+    LEFTARG = anyelement,
+    RIGHTARG = jsonb,
+    RESTRICT = search_config_restrict
+);
+
+CREATE OPERATOR pg_catalog.@@@ (
+    PROCEDURE = search_with_text,
+    LEFTARG = anyelement,
+    RIGHTARG = text,
+    RESTRICT = text_restrict
+);
+
 CREATE OPERATOR pg_catalog.@@@ (
-    PROCEDURE = search_tantivy,
+    PROCEDURE = search_with_query_input,
```

Comment:
FYI, this function name has changed since there's now 2.  one for plain text, one for paradedb.searchqueryinput (this one), and one for the SearchConfig jsonb version (which was previously the only case)